src/sentry/types/organization.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from functools import cached_property
+from typing import Optional
+
+from sentry import features
+from sentry.app import env
+from sentry.utils.http import is_using_customer_domain
+
+
+class OrganizationAbsoluteUrlMixin:
+    slug: str
+
+    @cached_property
+    def _has_customer_domain(self) -> bool:
+        """
+        Check if the current organization is using or has access to customer domains.
+        """
+
+        request = env.request
+        if request and is_using_customer_domain(request):
+            return True
+
+        return features.has("organizations:customer-domains", self)
```

Comment:
We'll need to make sure that the feature context generation is compatible with both Organization implementations going forwards. Perhaps we should add some tests to getsentry for that.